### PR TITLE
Support custom gates in Wasm bindings

### PR DESCRIPTION
**Pull request by rbonichon**
This adds the necessary code to support custom gates in wasm bindings

### CODE REVIEW

**Review by mitschabaude — CHANGES_REQUESTED** [review]
I have one important comment!

File: kimchi/wasm/src/lib.rs [hunk 1]
```diff
@@ -73,7 +74,7 @@ pub fn wait_until_non_zero(ptr: *const u32) -> u32 {
     unreachable!();
 }
 
-pub mod rayon;
+pub use wasm_bindgen_rayon::init_thread_pool;
```
[the remark this answers is not in the document]
Comment (rbonichon) [reply kimchi/wasm/src/lib.rs:77]: indeed. let me fix this.

[the remark this answers is not in the document]
Comment (rbonichon) [reply kimchi/wasm/src/lib.rs]: fixed!

**Review by mitschabaude — APPROVED** [review]

**Review by mrmr1993 — CHANGES_REQUESTED** [review]
Please don't mix in unrelated changes.

It looks like the JS side isn't complete, since it never calls the newly added accessors, or uses the new arguments. If you search for `emul_comm` in the JS code, you should find the relevant places.

**Comment by rbonichon** [comment]
> Please don't mix in unrelated changes.

Clippy suggestions and/or missing functions in JS bindings will be moved to other PRs

File: kimchi/js/bindings.js [hunk 2]
```diff
@@ -1424,6 +1447,12 @@ var caml_plonk_verification_evals_to_rust = function (x, klass, poly_comm_class,
         return caml_poly_comm_to_rust_poly_comm(poly_comm, poly_comm_class, mk_affine);
     };
 
+    var convert_opt = function (poly_comm_opt) {
```
Comment (mrmr1993) [inline kimchi/js/bindings.js:1450]: I don't think this does what we want. The JSOO representation of an OCaml `option` is `0` for `None` or `[0, foo]` for `Some foo`. You can see similar being done elsewhere in this file

Comment (rbonichon) [reply kimchi/js/bindings.js:1450]: Doesn't it? Don't we want to map `None` to `undefined` and `Some x` to `caml_poly_comm_to_rust_poly_comm(x, ....)`  in this `to_rust` function ?

Comment (rbonichon) [reply kimchi/js/bindings.js:1450]: AFAICT, the code here is very similar to `caml_poly_comm_to_rust_poly_comm` where we have the following snippet
```
    var caml_unshifted = poly_comm[1];
    var caml_shifted = poly_comm[2];
    var rust_shifted = undefined;
    if (caml_shifted !== 0) {
        rust_shifted = rust_affine_of_caml_affine(caml_shifted[1], mk_affine);
    }
```

**Review by mrmr1993 — APPROVED** [review]

**Pull request merged**